typing_extensions/src_py3/typing_extensions.py

```diff
@@ -0,0 +1,612 @@
+import abc
+import collections
+import contextlib
+import sys
+import typing
+try:
+    import collections.abc as collections_abc
+except ImportError:
+    import collections as collections_abc  # Fallback for PY3.2.
+
+if hasattr(typing, '_generic_new'):
+    _generic_new = typing._generic_new
+else:
+    def _generic_new(base_cls, cls, *args, **kwargs):
+        return base_cls.__new__(cls, *args, **kwargs)
```

Comment:
I don't know if I necessarily have tests for specifically this -- I was mostly copying what the typing module was already doing and assumed the behavior was correct. For example, in Python 3.5.0 - 3.5.2, the `List` type was defined like this:

```python
class List(list, MutableSequence[T], extra=list):
    def __new__(cls, *args, **kwds):
        if _geqv(cls, List):
            raise TypeError("Type List cannot be instantiated; "
                            "use list() instead")
        return list.__new__(cls, *args, **kwds)
```

In Python 3.5.3+, it was defined like this:

```python
class List(list, MutableSequence[T], extra=list):
    __slots__ = ()

    def __new__(cls, *args, **kwds):
        if _geqv(cls, List):
            raise TypeError("Type List cannot be instantiated; "
                            "use list() instead")
        return _generic_new(list, cls, *args, **kwds)
```

I'm also not sure if this substitution is 100% correct, but if it's broken, it'll at least be consistently broken in the same way the rest of the `typing` module was in Python 3.5.0 - 3.5.2. Defining the function this way also helped me reduce redundancy since I can define things like `Deque` just once instead of twice.